installation_and_upgrade/Jenkinsfile_vhd_creation

```diff
@@ -32,14 +32,7 @@ pipeline {
   }
 
   post {
-    failure {
-        bat """
-            set BUILD_NUMBER=${env.BUILD_NUMBER}
-            installation_and_upgrade/vhd_build_force_cleanup.bat
-            """
-    }
-
-    unstable {
```

Comment:
We've used `cleanup` rather than `always` in some other places, it has the advantage that it will still run after other posts allowing them to be added if needed.  I had a feeling `cleanup` might also run on job abort whereas `always` might need job to terminate, but i'm not sure. I think the problem there was that if you hit a complete job timeout then that also stopped post actions running,  so using stage timeout is best. I see this jenkinsfile has an overall timeout si i think it may also suffer that problem - we should move that to a stage timeout

Reply:
According to the docs runs "regardless of the completion status of the Pipeline’s or stage’s run". But it does seem better to do cleanup which runs after all the other post steps. In theory the post steps should run regardless of the completion status of the pipeline, but it doesn't hurt to move the timeouts so I have

Reply:
post steps can still take time, so they do seem to get covered by an overall job timeout as post steps can hang too. It did surprise me at the time, but it makes sense, and would tally with what I saw (a make clean did not get run when job timeout happened).  It also seems that if you had "always" and "success" steps then "always" gets run before "success" which may not be what you want, so I think "cleanup" is better for that reason too.

Reply:
The full job timeout would probably have been ok if its value had been > (sum of timeouts of other stage timeouts already there) so the post had some time to run